datafusion/sql/src/unparser/plan.rs

```diff
@@ -121,6 +124,40 @@ impl Unparser<'_> {
         }
     }
 
+    /// Try to unparse a [UserDefinedLogicalNode] to a SQL statement.
+    /// If multiple unparsers are registered for the same [UserDefinedLogicalNode],
+    /// the last unparsing result will be returned.
+    fn extension_to_statement(
+        &self,
+        node: &dyn UserDefinedLogicalNode,
+    ) -> Result<ast::Statement> {
+        let mut statement = None;
+        for unparser in &self.udlp_unparsers {
+            statement = unparser.unparse_to_statement(node, self)?;
+        }
+        if let Some(statement) = statement {
+            Ok(statement)
+        } else {
+            not_impl_err!("Unsupported extension node: {node:?}")
+        }
+    }
+
+    /// Try to unparse a [UserDefinedLogicalNode] to a SQL statement.
+    /// If multiple unparsers are registered for the same [UserDefinedLogicalNode],
+    /// all of them will be called in order.
+    fn extension_to_sql(
+        &self,
+        node: &dyn UserDefinedLogicalNode,
+        query: &mut Option<&mut QueryBuilder>,
+        select: &mut Option<&mut SelectBuilder>,
+        relation: &mut Option<&mut RelationBuilder>,
+    ) -> Result<()> {
+        for unparser in &self.udlp_unparsers {
```

Comment:
It might be good to add indication that `unparse` applied to be consistent with `unparse_to_statement` and throw error if non of registered udlps applied / successfully processed the node.

Reply:
Yeah, if none of the extension unparsers were able to process it, it should throw an error IMO

Reply:
Sound greats. I think we could provide some `enum` for it like
```
UnparsedResult::Unparsed()
UnparsedResult::Original()
..
```
I'll try it